dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -749,6 +752,402 @@ def test_min_max_id_properly_update(self, *mocks):
         self.assertEqual(6, profile._max_id)
 
 
+@mock.patch('dataprofiler.profilers.profile_builder.UnstructuredCompiler',
+            spec=UnstructuredCompiler)
+@mock.patch('dataprofiler.profilers.profile_builder.DataLabeler')
+class TestUnstructuredProfiler(unittest.TestCase):
+
+    @classmethod
+    def setUp(cls):
+        test_utils.set_seed(seed=0)
+
+    def test_base(self, *mocks):
+        # ensure can make an empty profiler
+        profiler = UnstructuredProfiler(None)
+        self.assertIsNone(profiler.encoding)
+        self.assertIsNone(profiler.file_type)
+        self.assertIsNone(profiler._profile)
+        self.assertIsNone(profiler._samples_per_update)
+        self.assertEqual(0, profiler._min_true_samples)
+        self.assertEqual(0, profiler.total_samples)
+        self.assertEqual(0, profiler._empty_line_count)
+        self.assertEqual(0.2, profiler._sampling_ratio)
+        self.assertEqual(5000, profiler._min_sample_size)
+        self.assertIsInstance(profiler.options, ProfilerOptions)
+
+        # can set samples_per_update and min_true_samples
+        profiler = UnstructuredProfiler(None, samples_per_update=10,
+                                        min_true_samples=5)
+        self.assertEqual(profiler._samples_per_update, 10)
+        self.assertEqual(profiler._min_true_samples, 5)
+
+        # can properties update correctly for data
+        data = pd.Series(['this', 'is my', '\n\r', 'test'])
+        profiler = UnstructuredProfiler(data)
+        self.assertEqual(4, profiler.total_samples)
+        self.assertEqual(1, profiler._empty_line_count)
+        self.assertEqual("<class 'pandas.core.series.Series'>",
+                         profiler.file_type)
+        self.assertIsNone(profiler.encoding)
+        self.assertIsInstance(profiler._profile, UnstructuredCompiler)
+
+    def test_merge_profiles(self, *mocks):
+        # can properties update correctly for data
+        data1 = pd.Series(['this', 'is my', '\n\r', 'test'])
+        data2 = pd.Series(['here\n', '\t    ', ' ', ' is', '\n\r', 'more data'])
+
+        # create profilers
+        profiler1 = UnstructuredProfiler(data1)
+        profiler2 = UnstructuredProfiler(data2)
+
+        # mock out _profile
+        profiler1._profile = 1
+        profiler2._profile = 2
+
+        # merge profilers
+        merged_profile = profiler1 + profiler2
+        self.assertEqual(10, merged_profile.total_samples)
+        self.assertEqual(4, merged_profile._empty_line_count)
+        # note how sample doesn't include whitespace lines
+        self.assertCountEqual(['this', ' is', 'here\n', 'more data', 'is my'],
```

Comment:
should this include the string `test`?

Reply:
> should this include the string test?
it won't have test because the merged samples is a random sampling between the two previous sets of samples, hence it maybe a mixture of both and not a concatenation

> can we add the case when is_case_sensitive is False, then the merging of 'Test' and 'test' will become only 'test'?
sample doesn't alter case, it takes the direct data

Reply:
sample doesn't alter case, it takes the direct data